readthedocs_ext/readthedocs.py

```diff
@@ -126,6 +149,45 @@ def rtd_render(self, template, render_context):
                                                         app.builder.templates)
 
 
+def generate_json_artifacts(app, pagename, templatename, context, doctree):
+    """
+    Generate JSON artifacts for each page.
+
+    This way we can skip generating this in other build step.
+    """
+    try:
+        if not app.config.rtd_generate_json_artifacts:
+            return
+        # We need to get the output directory where the docs are built
+        # _build/json.
+        build_json = os.path.abspath(
+            os.path.join(app.outdir, '..', 'json')
+        )
+        outjson = os.path.join(build_json, pagename + '.fjson')
+        outdir = os.path.dirname(outjson)
+        if not os.path.exists(outdir):
+            os.makedirs(outdir)
+        with open(outjson, 'w+') as json_file:
+            to_context = {
```

Comment:
i think the field should be blank rather than non existence if the key is missing.
It should be something like
```python
to_context = { key: context.get(key, '') for key in KEYS}
```

Reply:
This makes sense, so that the JSON always has the same keys.